src/java/org/apache/cassandra/io/sstable/format/big/BigTableWriter.java

```diff
@@ -405,13 +414,13 @@ public MmappedRegionsCache getMmappedRegionsCache()
         }
 
         @Override
-        public SequentialWriter getDataWriter()
+        public LazyCloseableRef<SequentialWriter> getDataWriter()
```

Comment:
I would switch the interface back to returning the objects, but create them here (if necessary, implemented as calling the reference's `get` method).